drone-localization/core/models/dual_branch_swin/loss.py

```
@@ -0,0 +1,23 @@
+    def __init__(self, m=0.25, gamma=256):
+        super().__init__()
+        self.m = m
+        self.gamma = gamma
+
+    def forward(self, feat, labels):
+        sim = torch.matmul(feat, feat.T)
+        lbl = labels.unsqueeze(0) == labels.unsqueeze(1)
+        pos_m = lbl & ~torch.eye(labels.size(0), dtype=torch.bool, device=feat.device)
+        neg_m = ~lbl
+        sp, sn = sim[pos_m], sim[neg_m]
+        if len(sp) == 0 or len(sn) == 0: return (feat * 0.0).sum()
```

Ruff (0.15.12)

[error] Multiple statements on one line (colon)

(E701)